pkg/geo/geomfn/subdivide.go

```diff
@@ -0,0 +1,364 @@
+			}
+		} else {
+			// Ignore point on the boundaries
+			if clip.LoY == pivot || clip.HiY == pivot {
+				pivot = (clip.LoY + clip.HiY) / 2
+			}
+		}
+	default:
+		if splitHorizontally {
+			pivot = (clip.LoX + clip.HiX) / 2
+		} else {
+			pivot = (clip.LoY + clip.HiY) / 2
+		}
+	}
+	return pivot, nil
+}
+
+// findPolygonsMostCentralPointValue finds the value of the most central point for a geom.Polygon.
+// Can be X or Y value, depending on the splitHorizontally argument.
+func findPolygonsMostCentralPointValue(
+	poly *geom.Polygon, nVertices int, splitHorizontally bool, clip *geo.CartesianBoundingBox,
+) (float64, error) {
+	pivot := math.MaxFloat64
+	ringToTrimIndex := 0
+	ringArea := float64(0)
+	pivotEps := math.MaxFloat64
+	var ptEps float64
+	nVerticesOuterRing, err := countVertices(poly.LinearRing(0))
+	if err != nil {
+		return 0, errors.Newf("error counting vertices: %v", err)
+	}
+	// if there are more points in holes than in outer ring
+	if nVertices >= 2*nVerticesOuterRing {
+		// trim holes starting from biggest
+		for i := 1; i < poly.NumLinearRings(); i++ {
+			currentRingArea := poly.LinearRing(i).Area()
+			if currentRingArea >= ringArea {
+				ringArea = currentRingArea
+				ringToTrimIndex = i
+			}
+		}
+	}
+	pa := poly.LinearRing(ringToTrimIndex)
+	// Find the most central point in the chosen ring
+	for j := 0; j < pa.NumCoords(); j++ {
+		var pt float64
+		if splitHorizontally {
+			pt = pa.Coord(j).X()
+			xHalf := (clip.LoX + clip.HiX) / 2
+			ptEps = math.Abs(pt - xHalf)
+		} else {
+			pt = pa.Coord(j).Y()
+			yHalf := (clip.LoY + clip.HiY) / 2
+			ptEps = math.Abs(pt - yHalf)
+		}
+		if ptEps < pivotEps {
+			pivot = pt
+			pivotEps = ptEps
+		}
+	}
+	return pivot, nil
+}
+
+// clipGeomTByBoundingBox returns a result of intersection and simplification of the geom.T and bounding box provided.
+// The result is also a geom.T.
+func clipGeomTByBoundingBox(gt geom.T, clip *geo.CartesianBoundingBox) (geom.T, error) {
+	g, err := geo.MakeGeometryFromGeomT(gt)
+	if err != nil {
+		return geom.NewGeometryCollection(), errors.Newf("error transforming geom.T to geometry: %v", err)
+	}
+	clipgt := clip.ToGeomT(g.SRID())
+	clipg, err := geo.MakeGeometryFromGeomT(clipgt)
+	if err != nil {
+		return geom.NewGeometryCollection(), errors.Newf("error transforming geom.T to geometry: %v", err)
+	}
+	out, err := Intersection(g, clipg)
+	if err != nil {
+		return geom.NewGeometryCollection(), errors.Newf("error applying intersection: %v", err)
+	}
+	// Simplify is required to remove the unnecessary points. Otherwise vertices count is altered and too many subdivision may be returned.
+	out, err = Simplify(out, 0)
+	if err != nil {
+		return geom.NewGeometryCollection(), errors.Newf("simplifying error: %v", err)
+	}
+	gt, err = out.AsGeomT()
+	if err != nil {
+		return geom.NewGeometryCollection(), errors.Newf("error transforming geometry to geom.T: %v", err)
+	}
+	return gt, nil
+}
+
+func countVertices(gt geom.T) (int, error) {
```

Comment:
this is a copy of https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/sem/builtins/geo_builtins.go#L1933 -- can you move the logic into here? i'm not fussed if you don't add any tests for that function into this package.

Reply:
Ok, moved. Is it ok to keep it in this file?

Reply:
i think we should move it to unary_operators.go

Reply:
Moved